odl/trafos/fourier.py

```diff
@@ -1645,6 +1647,211 @@ def inverse(self):
             sign=sign, tmp_r=self._tmp_r, tmp_f=self._tmp_f)
 
 
+class NonUniformFourierTransformBase(Operator):
+    """Non uniform Fast Fourier Transform.
+
+    The normalization is inspired from pysap-mri, mainly this class:
+    https://github.com/CEA-COSMIC/pysap-mri/blob/master/mri/reconstruct/fourier.py#L123
+    """
+    def __init__(self, shape, non_uniform_samples, domain, range):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        shape : tuple
+            The dimensions of the data whose non uniform FFT you want to
+            compute
+        non_uniform_samples : iterable
```

Comment:
Does this need to be array-like? Or is iterable enough?

Reply:
Yes it needs to be array-like, I will update that!